product-service/src/data-access/products.ts

```diff
@@ -1,16 +1,70 @@
+import pg from "pg";
 import { IProductHandler, Product } from "../utils/interfaces";
-import { Products } from "../utils/mockData";
+import dbCreds from "./dbCreds";
+
 
 class ProductHandler implements IProductHandler {
-    async getProduct(id: number): Promise<Product> {
-        return Products[id];
+    async getProduct(id: string): Promise<Product> {
+        const client = new pg.Client(dbCreds);      
+        try {
+            client.connect();
+            console.log(id);
+            const result = await client.query(
+                `SELECT products.id AS id, title, description, image, price * 0.01 AS price, 
+                    (SELECT count FROM stocks WHERE stocks.product_id = '${id}') AS count 
+                    FROM products WHERE products.id = '${id}';`
```

Comment:
In one of PR's that I have reviewed someone left this note. I guess, you have the same situation here
```
From security perspective it is safer to pass parameters as client.query('query string', [param1, param2]).
Injecting values directly into query string opens SQL injection vulnerabilities.
```